openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
 class YarnStatusGetter(JobMetadataGetterInterface):
     """YARN app status getter"""
 
+    def get_application_url(self, application_id: str) -> str:
+        return url_join(
+            ConfigParams().yarn_rest_api_base_url,
+            f"/ws/v1/cluster/apps/{application_id}",
+        )
+
+    def get_authentication_provider(self):
+        auth = None
+        if ConfigParams().yarn_auth_use_kerberos:
+            # TODO: HTTPKerberosAuth is deprecated, this is a utility class for compatibility with requests_kerberos.
+            #   Find the better way to do this, using requests_gssapi directly.
+            # Have to set mutual_authentication to REQUIRED or it just skips kerberos auth it seems.
+            auth = requests_gssapi.HTTPKerberosAuth(
+                mutual_authentication=requests_gssapi.REQUIRED
+            )
+        return auth
```

Comment:
If `yarn_auth_use_kerberos` is not necessary (see above) we can remove this method and simply initialize `auth`.

Reply:
Now the constructor accepts an optional auth object, which is None by default.
That has helped to simplify quite a bit.

- Basically, the config param for yarn_auth_use_kerberos and related code has been removed entirely. 
- Unit tests leave just out the auth param
- And only the CLI app passes the Kerberos auth to the YarnStatusGetter constructor here:
https://github.com/Open-EO/openeo-geopyspark-driver/blob/3920e20dda0b1ef027148d0146658b696c4c87a9/openeogeotrellis/job_tracker_v2.py#L537-L543